assets/src/application/services/utilities/useTimeout.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+
+/**
+ * @link https://github.com/bmcmahen/toasted-notes/blob/master/src/Alert/useTimeout.ts
+ */
+const useTimeout = (callback: VoidFunction, delay: number | null): void => {
+	const savedCallback = React.useRef<VoidFunction | null>();
+
+	// Remember the latest callback.
+	React.useEffect((): void => {
+		savedCallback.current = callback;
+	}, [callback]);
+
+	// Set up the interval.
+	React.useEffect(() => {
+		const tick = (): void => {
```

Comment:
We should stick to `import React, { useEffect } from 'react';`

Reply:
agree, updated